gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/FolderCompletionContributor.java

```diff
@@ -56,6 +58,25 @@ protected void addCompletions(
         });
     }
 
+    private void cacheCompletion(IElementType elementType) {
+        extend(CompletionType.BASIC, PlatformPatterns.psiElement(elementType), new CompletionProvider<>() {
+            @Override
+            protected void addCompletions(
+                    CompletionParameters parameters, ProcessingContext context, CompletionResultSet resultSet) {
+
+                List<String> repositories = List.of("https://repo.maven.apache.org/maven2/");
+
+                DependencyGroup group = DependencyGroup.groupFromParameters(parameters);
+                GradleCacheExplorer gradleCacheExplorer = new GradleCacheExplorer(repositories);
```

Comment:
what is the lifecycle of the enclosing method? is `addCompletions` going to be called on every completion suggestion? if so you'll end up gc-ing a lot and making a bunch of use once only `GradleCacheExplorer` objects. 

My gut feeling is that you *may* want `gradleCacheExplorer` to be a field in `FolderCompletionContributor`

Reply:
So I had the actual caffine cache as static in the object but this makes it very unclear so I have now made `gradleCacheExplorer` a field in `FolderCompletionContributor` which reads much better